bitsail-components/bitsail-component-metrics/bitsail-component-metrics-prometheus/src/main/java/com/bytedance/bitsail/component/metrics/prometheus/AbstractPrometheusReporter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 Bytedance Ltd. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.bytedance.bitsail.component.metrics.prometheus;
+
+import com.bytedance.bitsail.base.metrics.MetricManager;
+import com.bytedance.bitsail.base.metrics.MetricReporter;
+
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricRegistry;
+import io.prometheus.client.CollectorRegistry;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.regex.Pattern;
+
+@Slf4j
+public abstract class AbstractPrometheusReporter implements MetricReporter {
```

Comment:
why we need an abstract class here, this looks very simple

Reply:
This the common function use by `PrometheusMetricReporter`(pull mode) and `PrometheusPushGatewayReporter`(push mode) class, so I use a abstract class to do it, when we don't use dropwizard.metrics as the metrics core in bitsail, it will be complex.